hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/BaseHeapData.java

```diff
@@ -0,0 +1,184 @@
+/*
```

Comment:
Keep in mind that by introducing a new class of heap data, you could make it much more difficult for the JIT to do optimizations.